### DISCUSSION TITLE
Submit to remote Linux node with pyFluent

### DISCUSSION THREAD

**Discussion by fveick** [ansys/pyfluent#3544] [Q&A] [answered]
Hi,

I want to start a Fluent session on a remote linux node with my python code running from a Windows VDI.

I can to this via commandline with or without Slurm queue:

`fluent 3d -g -rsh="plink -batch -i private_key.ppk -l testuser -v" -nodepath=/ansys_inc/v242/fluent -node0=10.0.23.135 -setenv="FLUENT_REMOTE_SSH=1" -scheduler=slurm -scheduler_account=testuser -nm`
 
When I translate this to a launch_fluent command:

```
slurm = pyfluent.launch_fluent(precision="double",
                               processor_count=2,
                               mode="solver",
                               product_version='24.2',
                               additional_arguments='-rsh="plink -batch -i private_key.ppk -l testuser -N -v" -nodepath=/ansys_inc/v242/fluent -node0=10.0.23.135 -setenv="FLUENT_REMOTE_SSH=1"'
                               )
```
I am unable to spawn the processes on the Linux node, the processes on the Windows machine just seem to idle:

> Reading key file "private_key.ppk"
> Using username "testuser".
> Offered public key
> Offer of public key accepted
> Authenticating with public key "rsa-key-20241111"
> Sent public key signature
> Access granted
> Remote debug message: /home/testuser/.ssh/authorized_keys:1: key options: agent-forwarding port-forwarding pty user-rc x11-forwarding
> Remote debug message: /home/testuser/.ssh/authorized_keys:1: key options: agent-forwarding port-forwarding pty user-rc x11-forwarding
> Check your network connections and remote shell configuration.
> If you are on a slow network or running through a scheduler like lsf or sge, you may choose to wait.
> Type Ctrl-C to interrupt.

Any advice would be welcome

Best Regards
Farai

**Comment by rachelzhang333** [comment, marked as answer]
Do you consider to launch a Fluent session in Linux node first, then use 'pyfluent.connect_to_fluent' to connect to it?

**Reply by fveick** [reply]
Yes, that actually works and is the current workaround. The issue is just the launch from pyFluent itself.
